### PR TITLE
screensaver state support

### PR DESCRIPTION
**Pull request by nmfaraujo**

### CODE REVIEW

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java [hunk 1]
```diff
@@ -215,6 +217,7 @@ public void onClose(WebSocket webSocket) {
 		@Override
 		@SuppressWarnings("unchecked")
 		public void onMessage(String message) {
+			System.out.println(message);
```
Comment (avdleeuw) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java]: This println shouldn't be here, really

Comment (nmfaraujo) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java]: sorry, my first commit in git, i'll remove that

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import org.openhab.binding.xbmc.rpc.RpcCall;
+
+import com.ning.http.client.AsyncHttpClient;
+
+/**
+ * XBMC.GetInfoBooleans RPC
+ * 
+ * @author Nuno Araujo
+ * @since 1.5.0
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java]: please change to 1.7.0

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		Map<String, Object> result = getMap(response, "result");
+		if (result.containsKey("System.ScreenSaverActive")) {
+			Object o = result.get("System.ScreenSaverActive");
+			if(o!=null  && o instanceof Boolean)
+			{
+				screenSaverActive=(Boolean)o;
+			}
+		}
+			
+	}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java]: please format using the Eclipse Formatter (CTRL+SHFT+O)

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	@Override
+	protected String getName() {
+		return "XBMC.GetInfoBooleans";
+	}
+
+	@Override
+	protected Map<String, Object> getParams() {
+		List<String> properties = new ArrayList<String>();
+		properties.add("System.ScreenSaverActive");
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java]: please extract as constant

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		
+		Map<String, Object> params = new HashMap<String, Object>();
+		params.put("booleans", properties);
+		return params;
+	}
+
+	@Override
+	protected void processResponse(Map<String, Object> response) {
+		Map<String, Object> result = getMap(response, "result");
+		if (result.containsKey("System.ScreenSaverActive")) {
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/XBMCGetInfoBooleans.java]: please extract as constant

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java [hunk 6]
```diff
@@ -324,6 +329,16 @@ public void run() {
 			}
 		});
 	}
+	
+	private void updateScreenSaverStatus(boolean screenSaverActive) {
+		if (screenSaverActive) {
+			updateProperty("Screensaver.State", OnOffType.ON);
+		} else {
+			updateProperty("Screensaver.State", OnOffType.OFF);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java]: please extract key as constant

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java [hunk 7]
```diff
@@ -472,6 +487,16 @@ private void processSystemStateChanged(String method, Map<String, Object> json)
 			updateProperty("System.State", OnOffType.OFF);
 		} 	
 	}
+	
+	private void processScreensaverStateChanged(String method, Map<String, Object> json) {
+		System.out.println(method);
+		if ("GUI.OnScreensaverDeactivated".equals(method)) {
+			updateScreenSaverStatus(false);
+		}else if ("GUI.OnScreensaverActivated".equals(method)) {
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java]: please extract key as constant

File: bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java (hunk 1 above)
Comment (teichsta) [inline bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/XbmcConnector.java]: definitely … please use proper logging

**Comment by teichsta** [comment]
Thanks for this contribution, @nmfaraujo. Please find my the review comments. Also please squash your commits into one after incorporating the Review. Thanks, Thomas E.-E.

**Comment by nmfaraujo** [comment]
to use this new features, you have to add to items file, for example:

# this is for screensaver state
Switch ScreensaverSala "Screensaver Sala" {xbmc="<[#livingRoom|Screensaver.State]"}

# this is for opening a tv channel by name
String XBMC_OpenTVChannel    "XBMC Open TV Channel [%s]"  { xbmc=">[#livingRoom|PVR.OpenTV]" }

# this is for opening a radio channel by name
String XBMC_OpenRadioChannel    "XBMC Open Radio Channel [%s]"  { xbmc=">[#livingRoom|PVR.OpenRadio]" }

a sample rule that uses screensaver state:

rule "turn tv on or off"
        when
                Item ScreensaverSala changed
        then
                if(ScreensaverSala.state == ON)
                {
                        logInfo("screensaver", "turn tv OFF")
                        sendCommand(TV_PowerSwitch,OFF)
                }
                else if(ScreensaverSala.state == OFF)
                {
                        logInfo("screensaver", "turn tv ON")
                        sendCommand(TV_PowerSwitch,ON)
                }
end

to open a tv channel:

// the command is the name of the channel 
sendCommand(XBMC_OpenTVChannel, "Discovery Channel")

**Pull request merged**